test/e2e/query_test.go

```diff
@@ -1006,3 +1004,132 @@ func synthesizeSamples(ctx context.Context, prometheus *e2e.InstrumentedRunnable
 	compressed := snappy.Encode(buf, pBuf.Bytes())
 	return client.Store(ctx, compressed)
 }
+
+func TestSidecarQueryEvaluationWithDedup(t *testing.T) {
+	t.Parallel()
+
+	ts := []struct {
+		prom1Samples []fakeMetricSample
+		prom2Samples []fakeMetricSample
+		query        string
+		result       model.Vector
+	}{
+		{
+			query:        "max (my_fake_metric)",
+			prom1Samples: []fakeMetricSample{{"i1", 1}, {"i2", 5}, {"i3", 9}},
+			prom2Samples: []fakeMetricSample{{"i1", 3}, {"i2", 4}, {"i3", 10}},
+			result: []*model.Sample{
+				{
+					Metric: map[model.LabelName]model.LabelValue{},
+					Value:  9, // Why? Very weird.
```

Comment:
I suppose dedup picks up the first seriesSet instead of the second one and then just goes on with it even though without pushdown we would compute max() over the whole data set?